### DISCUSSION TITLE
Getting "session expired" even with App Store Connect API Key

### DISCUSSION THREAD

**Discussion by o15a3d4l11s2** [fastlane/fastlane#20466] [Q&A]
I cannot seem to understand how is it possible that although I am using App Store Connect API Key, occasionally I still get `"Your session has expired. Please log in."`. Could it be related to me logging in in App Store Connect web interface in the same time while submitting app through Fastlane? Or could it be related to us using it through Jenkins to submit multiple apps in a short period of time and getting limited by Apple?

**Update**
Here is my lane
```
lane :distribute_to_app_store do
  app_store_connect_api_key
  build_app
  upload_to_app_store
end
```

**Update 2**
I also have an `Appfile` that was used before adding the `app_store_connect_api_key` and it still contains
```
app_identifier('XXXXXX')
apple_id('XXXXXX')
itc_team_id('XXXXX')
team_id('XXXXX')
```
I start thinking that there might be a mix between the api key and the old way of authenticating (maybe when the api key expires after 500 seconds).

**Comment by KevinKreps** [comment]
@o15a3d4l11s2 Facing a similar issue, have you found the cause for yours?

**Reply by KevinKreps** [reply]
Logging in to https://developer.apple.com/ solves this problem

**Reply by o15a3d4l11s2** [reply]
The actual problem is that it should use API key and not a session, so my issue is still valid. Happy that you were able to work around it by logging in.
